godot-core/src/builtin/color_hsv.rs

```diff
@@ -0,0 +1,247 @@
+///  
+/// ## Reference
+/// - Smith, Alvy Ray. "Color gamut transform pairs." ACM Siggraph Computer Graphics 12.3 (1978): 12-19.
+#[derive(Debug, Clone, Copy, PartialOrd, Default)]
+pub struct ColorHsv {
+    h: f32,
+    s: f32,
+    v: f32,
+    a: f32,
+}
+
+// Manual implementation because Hue minimal value (0.) is equal to maximal value (1.)
+impl PartialEq for ColorHsv {
+    fn eq(&self, other: &Self) -> bool {
+        (self.h == other.h || (self.h - other.h).abs() == 1.)
+            && self.s == other.s
+            && self.v == other.v
+            && self.a == other.a
+    }
+}
+
+impl ColorHsv {
+    /// Construct from Hue, Saturation, Value where all values are between `0.0` and `1.0`.
+    ///
+    /// If values outside of that range are provided, they will be clamped to valid range. To add `alpha` value, use  `with_alpha()` method.
+    pub fn from_hsv(h: f32, s: f32, v: f32) -> Self {
+        ColorHsv {
+            h: h.clamp(0., 1.),
+            s: s.clamp(0., 1.),
+            v: v.clamp(0., 1.),
+            a: 1.,
+        }
+    }
+
+    /// Construct from Hue, Saturation, Value where Hue is represented in deegrees (`0.0` - `360.0`).
+    ///
+    /// Saturation, Value and Alpha have to be provided in range between `0.0` an `1.0`. If values outside of valid ranges are provided,
+    /// they will be clamped to a valid range. To add `alpha` value, use `with_alpha()` method.
+    pub fn from_hsv_deg(h: f32, s: f32, v: f32) -> Self {
+        ColorHsv {
+            h: h.clamp(0., 360.) / 360.,
+            s: s.clamp(0., 1.),
+            v: v.clamp(0., 1.),
+            a: 1.,
+        }
+    }
+
+    /// Adds an alpha value.
+    pub fn with_alpha(mut self, alpha: f32) -> Self {
+        self.a = alpha.clamp(0., 1.);
+        self
+    }
+
+    pub fn h(&self) -> f32 {
+        self.h
+    }
+
+    pub fn h_deg(&self) -> f32 {
+        self.h * 360.
+    }
+
+    pub fn set_h(&mut self, h: f32) {
+        self.h = h.clamp(0., 1.);
+    }
+
+    pub fn set_h_deg(&mut self, h: f32) {
+        self.h = h.clamp(0., 360.) / 360.;
+    }
+
+    pub fn s(&self) -> f32 {
+        self.s
+    }
+
+    pub fn set_s(&mut self, s: f32) {
+        self.s = s.clamp(0., 1.);
+    }
+
+    pub fn v(&self) -> f32 {
+        self.v
+    }
+
+    pub fn set_v(&mut self, v: f32) {
+        self.v = v.clamp(0., 1.);
+    }
+
+    pub fn alpha(&self) -> f32 {
+        self.a
+    }
+
+    pub fn set_alpha(&mut self, alpha: f32) {
+        self.a = alpha.clamp(0., 1.);
+    }
+}
+
+impl From<Color> for ColorHsv {
+    fn from(value: Color) -> Self {
+        let (h, s, v, a) = rgb_to_hsv_f(value.r, value.g, value.b, value.a);
+
+        ColorHsv { h, s, v, a }
+    }
+}
+
+impl From<ColorHsv> for Color {
+    fn from(value: ColorHsv) -> Self {
+        let (r, g, b, a) = hsv_to_rgb_f(value.h, value.s, value.v, value.a);
+
+        Color { r, g, b, a }
+    }
+}
+
+fn rgb_to_hsv_f(r: f32, g: f32, b: f32, a: f32) -> (f32, f32, f32, f32) {
```

Comment:
Probably `rgba_to_hsva` is more descriptive.

Why the `f`? float? Is it really needed, as there is no other method with a different representation?

Reply:
Yeah, it was left from the two different implementations as I've described in the initial comment (`f` and `i`). Will remove, good catch!